coil/src/main/java/com/google/accompanist/coil/Coil.kt

```diff
@@ -68,313 +61,96 @@ object CoilImageDefaults {
-        content = content
+        context = context,
     )
+}.apply {
+    this.data = data
+    this.requestBuilder = requestBuilder
 }
 
+private typealias RequestBuilder = (ImageRequest.Builder.(size: IntSize) -> ImageRequest.Builder)
+
 /**
- * Creates a composable that will attempt to load the given [request] using [Coil], and provides
- * complete content of how the current state is displayed:
+ * A state object that can be hoisted for [com.google.accompanist.imageloading.AsyncImage]
+ * to load images using [coil.Coil].
  *
- * ```
- * CoilImage(
- *   request = ImageRequest.Builder(context).data(...).build(),
- * ) { imageState ->
- *   when (imageState) {
- *     is ImageLoadState.Success -> // TODO
- *     is ImageLoadState.Error -> // TODO
- *     ImageLoadState.Loading -> // TODO
- *     ImageLoadState.Empty -> // TODO
- *   }
- * }
- * ```
+ * In most cases, this will be created via [rememberCoilAsyncImageState].
  *
- * @param request The request to execute. If the request does not have a [ImageRequest.sizeResolver]
- * set, one will be set on the request using the layout constraints.
- * @param modifier [Modifier] used to adjust the layout algorithm or draw decoration content.
- * @param requestBuilder Optional builder for the [ImageRequest].
  * @param imageLoader The [ImageLoader] to use when requesting the image. Defaults to
  * [CoilImageDefaults.defaultImageLoader].
- * @param previewPlaceholder Drawable resource ID which will be displayed when this function is
- * ran in preview mode.
- * @param shouldRefetchOnSizeChange Lambda which will be invoked when the size changes, allowing
- * optional re-fetching of the image. Return true to re-fetch the image.
- * @param onRequestCompleted Listener which will be called when the loading request has finished.
- * @param content Content to be displayed for the given state.
+ * @param context The Android [Context] to use when creating [ImageRequest]s.
  */
-@Composable
-fun CoilImage(
-    request: ImageRequest,
-    modifier: Modifier = Modifier,
-    requestBuilder: (ImageRequest.Builder.(size: IntSize) -> ImageRequest.Builder)? = null,
-    imageLoader: ImageLoader = CoilImageDefaults.defaultImageLoader(),
-    @DrawableRes previewPlaceholder: Int = 0,
-    shouldRefetchOnSizeChange: (currentResult: ImageLoadState, size: IntSize) -> Boolean = DefaultRefetchOnSizeChangeLambda,
-    onRequestCompleted: (ImageLoadState) -> Unit = EmptyRequestCompleteLambda,
-    content: @Composable BoxScope.(imageLoadState: ImageLoadState) -> Unit
-) {
-    if (LocalInspectionMode.current && previewPlaceholder != 0) {
-        // If we're in inspection mode (preview) and we have a preview placeholder, just draw
-        // that using an Image and return
-        Image(
-            painter = painterResource(previewPlaceholder),
-            contentDescription = null,
-            modifier = modifier,
-            contentScale = ContentScale.Crop,
-        )
-        return
-    }
+@Stable
```

Comment:
Is this really stable or are we using this to say ignore changes to these params?

Reply:
Ja, it's really stable.